src/main/java/com/google/opensesame/servlets/MentorsServlet.java

```diff
@@ -26,14 +29,34 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws ServletException, IOException {
-    addMockMentor(response); // TODO: remove by production
     List<UserEntity> mentorEntities =
         ofy().load().type(UserEntity.class).filter("isMentor", true).list();
+    System.out.println(mentorEntities);
     ArrayList<UserData> mentors = new ArrayList<UserData>();
     for (UserEntity mentorEntity : mentorEntities) {
       mentors.add(new UserData(mentorEntity));
     }
 
+    // If the user is logged in and has a profile, sort mentors by common interest.
+    User loggedInUser = AuthServlet.getAuthorizedUser();
+    if (loggedInUser != null) {
+      String userID = loggedInUser.getUserId();
+      UserEntity curUser = ofy().load().type(UserEntity.class).id(userID).now();
+      if (curUser != null) {
+        final UserData curUserData = new UserData(curUser);
+        Collections.sort(
+            mentors,
+            new Comparator<UserData>() {
+              @Override
+              public int compare(UserData u1, UserData u2) {
+                Integer u1Compatibility = new Integer(u1.compatibility(curUserData));
+                Integer u2Compatibility = new Integer(u2.compatibility(curUserData));
```

Comment:
This _might_ fall into premature optimization, but it might be worth pre-computing the compatibility scores of each of the mentors before hand and storing them in a HashMap so it doesn't need to be recomputed for every comparison.

Reply:
I think that is a premature optimization and we should revisit it later if necessary, but not include it in this PR.